boards/esp32-ttgo-lora32-v1/include/periph_conf.h

```diff
@@ -0,0 +1,203 @@
+#endif
+#ifndef I2C0_SDA
+#define I2C0_SDA    GPIO4             /**< SDA signal of I2C_DEV(0) [UEXT1] */
+#endif
+/** @} */
+
+/**
+ * @name   PWM channel configuration
+ *
+ * For generic boards, two PWM devices are configured. These devices
+ * contain all GPIOs that are not defined as I2C, SPI or UART for this board.
+ * Generally, all outputs pins could be used as PWM channels.
+ *
+ * @note As long as the according PWM device is not initialized with
+ * the ```pwm_init```, the GPIOs declared for this device can be used
+ * for other purposes.
+ *
+ * @{
+ */
+
+/**
+ * @brief Declaration of the channels for device PWM_DEV(0),
+ *        at maximum six channels.
+ */
+#ifndef PWM0_GPIOS
+#define PWM0_GPIOS  { GPIO0, GPIO17 }
+#endif
+
+/**
+ * @brief Declaration of the channels for device PWM_DEV(1),
+ *        at maximum six channels.
+ */
+#ifndef PWM1_GPIOS
+#define PWM1_GPIOS  { GPIO27  }
+#endif
+/** @} */
+
+/**
+ * @name    SPI configuration
+ *
+ * @note The GPIOs listed in the configuration are first initialized as SPI
+ * signals when the corresponding SPI interface is used for the first time
+ * by either calling the ```spi_init_cs``` function or the ```spi_acquire```
+ * function. That is, they are not allocated as SPI signals before and can
+ * be used for other purposes as long as the SPI interface is not used.
+ * @{
+ */
+#ifndef SPI0_CTRL
+#define SPI0_CTRL   VSPI    /**< VSPI is used as SPI_DEV(0) */
+#endif
+#ifndef SPI0_SCK
+#define SPI0_SCK    GPIO5   /**< VSPI SCK */
+#endif
+#ifndef SPI0_MISO
+#define SPI0_MISO   GPIO19  /**< VSPI MISO */
+#endif
+#ifndef SPI0_MOSI
+#define SPI0_MOSI   GPIO27  /**< VSPI MOSI */
+#endif
+#ifndef SPI0_CS0
+#define SPI0_CS0    GPIO18   /**< VSPI CS0 */
+#endif
+
+//#ifndef SPI1_DEV
+//#define SPI1_DEV    HSPI    /**< HSPI is used as SPI_DEV(1) */
+//#endif
+//#ifndef SPI1_SCK
+//#define SPI1_SCK    GPIO14  /**< HSPI SCK */
+//#endif
+//#ifndef SPI1_MISO
+//#define SPI1_MISO   GPIO12  /**< HSPI MISO */
+//#endif
+//#ifndef SPI1_MOSI
+//#define SPI1_MOSI   GPIO13  /**< HSPI MOSI */
+//#endif
+//#ifndef SPI1_CS0
+//#define SPI1_CS0    GPIO15  /**< HSPI CS0 */
+//#endif
+/** @} */
+
+/**
+ * @name   UART configuration
+ *
+ * ESP32 provides 3 UART interaces at maximum:
+ *
+ * UART_DEV(0) uses fixed standard configuration.<br>
+ * UART_DEV(1) is defined here.<br>
+ * UART_DEV(2) is not used.<br>
+ *
+ * @{
+ */
+#define UART0_TXD   GPIO10 /**< direct I/O pin for UART_DEV(0) TxD, can't be changed */
+#define UART0_RXD   GPIO9  /**< direct I/O pin for UART_DEV(0) RxD, can't be changed */
+
+#if FLASH_MODE_DOUT || FLASH_MODE_DIO || DOXYGEN
+#ifndef UART1_TXD
+#define UART1_TXD   GPIO10  /**< direct I/O pin for UART_DEV(1) TxD */
+#endif
+#ifndef UART1_RXD
+#define UART1_RXD   GPIO9   /**< direct I/O pin for UART_DEV(1) RxD */
+#endif
+#else
+#warning Configuration problem: Flash mode is qio or qout, \
+         GPIO9 and GPIO10 are not available for UART1 as configured
+#endif
+/** @} */
+
```

Comment:
According to this [pinout](https://primalcortex.files.wordpress.com/2017/11/ttgolorapinout_v2.jpg) UART2 is on pins GPIO16 and GPIO17. UART1 is not available.